### PR TITLE
Safe cancel

### PR DESCRIPTION
**Pull request by boeschf**

### CODE REVIEW

File: include/ghex/transport_layer/ucx/request.hpp [hunk 1]
```diff
@@ -272,38 +279,69 @@ namespace gridtools{
...
+                        //        // if we assume that the callback is always called, we
+                        //        // can handle this in the callback body- otherwise needs
+                        //        // to be done here:
+                        //        //request_init(ucx_ptr);
+                        //        ucp_request_cancel(worker, ucx_ptr);
+                        //    }
+                        //    m_req = nullptr;
+                        //    m_completed.reset();
+                        //    return true;
+                        //});
```
Comment (mbianco) [inline include/ghex/transport_layer/ucx/request.hpp]: Is this part relevant?

File: include/ghex/transport_layer/ucx/request.hpp [hunk 2]
```diff
@@ -272,38 +279,69 @@ namespace gridtools{
...
+
+                            if (m_req->m_kind == request_kind::send) return false;
+                            
+                            // set flag to cancelled
+                            *m_completed = 3;
+                                
+                            auto ucx_ptr = m_req->m_ucx_ptr;
+                            auto worker = m_req->m_worker->get();
+
+                            // set to zero here????
```
Comment (mbianco) [inline include/ghex/transport_layer/ucx/request.hpp]: What the status on this?

File: include/ghex/transport_layer/ucx/communicator.hpp [hunk 3]
```diff
@@ -351,7 +352,7 @@ namespace gridtools {
                         }
                         // else: cancelled - do nothing - cancel for sends does not exist
                         // set completion bit
-                        *req.m_completed = true;
+                        *req.m_completed = 2;
```
[the remark this answers is not in the document]
Comment (boeschf) [reply include/ghex/transport_layer/ucx/communicator.hpp]: yes I should use an `enum class` - will add it

File: include/ghex/transport_layer/ucx/request.hpp (hunk 1 above)
Comment (angainor) [reply include/ghex/transport_layer/ucx/request.hpp]: at least the comment is - we can't cancel send requests

**Pull request closed without merging**